Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -3690,11 +3691,26 @@ template <class TreeBuilder> TreeStatement Parser<LexerType>::parseImportDeclara
         }
 
         failIfFalse(autoSemiColon(), "Expected a ';' following a targeted import declaration");
-        return context.createImportDeclaration(importLocation, specifierList, moduleName, attributesList);
+        return context.createImportDeclaration(importLocation, type, specifierList, moduleName, attributesList);
     }
 
     bool isFinishedParsingImport = false;
-    if (matchSpecIdentifier()) {
+    if (Options::useImportDefer() && matchContextualKeyword(m_vm.propertyNames->deferKeyword)) {
+        // import defer NameSpaceImport FromClause ;
+        // import defer FromClause ;
+        JSTokenLocation deferLocation(tokenLocation());
+        next();
+        if (match(TIMES))
+            type = ImportDeclarationNode::ImportType::Deferred;
+        else {
+            auto specifier = context.createImportSpecifier(deferLocation, m_vm.propertyNames->defaultKeyword, m_vm.propertyNames->deferKeyword);
+            context.appendImportSpecifier(specifierList, specifier);
+            if (match(COMMA))
+                next();
+            else
+                isFinishedParsingImport = true;
```

Comment:
This looks not correct, as it is not corresponding to what `parseImportClauseItem` is doing.
We should not expand the code here, and we should use `parseImportClauseItem` instead.
Please use savePoint and restoreSavePoint mechanism when `defer` is detected. And doing special thing only when you find `defer` and `*`, and otherwise, let's just restore and going to the normal path following.

Reply:
Thanks for the feedback, just pushed a version that uses save points instead.